cherab/tools/observers/bolometry.py

```diff
@@ -687,6 +695,350 @@ def etendue_single_run(_):
+            [[pixel.centre_point.vector_to(self._slit.centre_point) for pixel in pixel_column]
+             for pixel_column in self.pixels_as_foils],
+            dtype='object'
+        )
+
+    @property
+    def slit(self):
+        return self._slit
+
+    @property
+    def curvature_radius(self):
+        return self._curvature_radius
+
+    @property
+    def units(self):
+        return self._units
+
+    @units.setter
+    def units(self, units):
+        if units == "Power":
+            pipeline = PowerPipeline2D(accumulate=self.accumulate)
+        elif units == "Radiance":
+            pipeline = RadiancePipeline2D(accumulate=self.accumulate)
+        else:
+            raise ValueError("The units property of BolometerFoil must be one of 'Power' or 'Radiance'.")
+        self._units = units
+        self.pipelines = [pipeline]
+
+    @property
+    def accumulate(self):
+        return self._accumulate
+
+    @accumulate.setter
+    def accumulate(self, value):
+        for pipeline in self.pipelines:
+            pipeline.accumulate = value
+            # Discard any samples from previous accumulate behaviour
+            pipeline.value.clear()
+
+    def as_sightlines(self):
+        """
+        Constructs a SightLine observer for each pixel in this bolometer.
+
+        :return: A 2D array of Sightline objects.
+        """
+
+        if self.units == "Power":
+            pipeline = PowerPipeline2D
+        else:
+            pipeline = RadiancePipeline2D
+
+        sightlines = []
+        for pixel_column in self.pixels_as_foils:
+            sightline_column = []
+            for pixel in pixel_column:
+                los_observer = SightLine(pipelines=[pipeline(accumulate=False)],
+                                         pixel_samples=1, quiet=True,
+                                         parent=self, name=pixel.name + "Sightline")
+                los_observer.render_engine = self.render_engine
+                los_observer.spectral_bins = self.spectral_bins
+                los_observer.min_wavelength = self.min_wavelength
+                los_observer.max_wavelength = self.max_wavelength
+                sightline_column.append(los_observer)
+            sightlines.append(sightline_column)
+
+        return np.asarray(sightlines, dtype='object')
+
+    def trace_sightlines(self, rows="all", columns="all"):
+        """
+        Traces the central sightlines through each pixel in the detector to see where the sightline terminates.
+
+        Raises a RuntimeError exception if no intersections were found.
+
+        :param rows: which pixel rows to trace. Can be an integer, a
+          list of integers or "all".
+        :param columns: which pixel columns to trace. Can be an
+          integer, a list of integers or "all".
+
+        :return: A 2D array of tuples containing the origin point, hit
+          point and terminating surface primitive for each pixel.
+        """
+        if rows == "all":
```

Comment:
What was wrong with None as a wildcard? :)

Reply:
To me, `rows=None` implies NO rows are returned. There's a second bit of logic a user must apply: "obviously that is stupid, so None must mean 'all'", so why not just state that explicitly?

Reply:
I can see your logic, but it does lead to unnecessary code. I feel it would be better to define in the docstring that it'll use all rows and columns unless explicit rows or columns are specified. Also mutable types should not be used as defaults. It's a bit of a style preference thing, so only a minor comment.

Reply:
OK, I remember now why I did this: bug #171 was causing very long run times for each pixel, so I wanted the caller to be able to restrict the number of pixels being traced to just the ones of interest. That bug was fixed by #172 and the sightline tracing is now very fast, so it's possible to just trace all the sightlines anyway and let the caller index the returned array for the lines they're actually interested in. For simplicity, and also for consistency with the `pixels_as_foils` and `as_sightlines` methods, I'll remove the `rows` and `columns` options entirely from this method.